test/integration/banListTest.ts

```diff
@@ -225,3 +227,63 @@ describe('Test: We will not be able to ban ourselves via ACL.', function () {
         assert.equal(acl.literalAclContent().deny.length, 3);
     })
 })
+
+
+describe.only('Test: ACL updates will batch when rules are added in succession.', function () {
+    it('Will batch ACL updates if we spam rules into a BanList', async function () {
+        this.timeout(180000)
+        const mjolnir = config.RUNTIME.client!
+        const serverName: string = new UserID(await mjolnir.getUserId()).domain
+        const moderator = await newTestUser({ name: { contains: "moderator" }});
+        moderator.joinRoom(this.mjolnir.managementRoomId);
+        const mjolnirId = await mjolnir.getUserId();
+        // How many rooms are we going to protect? Do we need to protect a few and verify all the ACL updates were batched in each?
+        const protectedRooms: string[] = [];
+        for (let i = 0; i < 10; i++) {
+            const room = await moderator.createRoom({ invite: [mjolnirId]});
+            await mjolnir.joinRoom(room);
+            await moderator.setUserPowerLevel(mjolnirId, room, 100);
+            await this.mjolnir!.addProtectedRoom(room);
+            protectedRooms.push(room);
+        }
+
+        // If a previous test hasn't cleaned up properly, these rooms will be populated by bogus ACLs at this point.
+        await this.mjolnir!.syncLists();
+        await Promise.all(protectedRooms.map(async room => {
+            // We're going to need timeline pagination I'm afraid.
+            const roomAcl = await mjolnir.getRoomStateEvent(room, "m.room.server_acl", "");
+            assert.equal(roomAcl?.deny?.length ?? 0, 0, 'There should be no entries in the deny ACL.');
+        }));
+
+        // Flood the subsribed list with banned servers, which should prompt Mjolnir to update server ACL in protected rooms.
+        const banListId = await moderator.createRoom({ invite: [mjolnirId] });
+        mjolnir.joinRoom(banListId);
+        this.mjolnir!.watchList(Permalinks.forRoom(banListId));
+        const acl = new ServerAcl(serverName).denyIpAddresses().allowServer("*");
+        for (let i = 0; i < 200; i++) {
+            acl.denyServer(i.toString());
```

Comment:
Nit: Rather than having `i.toString()` here and `${i}` below, could you define
```ts
const badServer = `${i}.evil.com`;
acl.denyServer(badServer);
await createPolicyRule(..., badServer, ...);
```

The more explicit name will make logs easier to read and reusing the variable makes the code clearer.